### PR TITLE
fix: update permissions for merge of products and versions

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-2821

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
     @Override
     public boolean hasAccess(Object obj) {
-        return false;
+        if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
+            return true;
+        } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
+            try {
+                // Need to make sure developer for products to be merged is active
+                List<Long> productIds = (List<Long>) obj;
+                // All products will have the same developer, so we only need to check the first one
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java]: I guess I'm just wondering if this comment is true... I think probably the way the UI lets you do a product merge it is true but if I just call the API with some random product ids what would happen?

Comment (andlar) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java]: Good catch Katy. Definitely see that as a possible issue

Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java]: I'll add to check each developer for a product.

Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java]: Sounds good.. make sure to do the same thing for the versions too as I think it had a similar comment.

**Pull request merged**